src/controllers/report-controller.js

```diff
@@ -0,0 +1,133 @@
+const Reports = require('../models/reports');
+const {StatusCodes} = require ('http-status-codes');
+const mongoose = require('mongoose');
+const {responseServer, responseNotFound} = require('../helpers/response-result');
+
+const postReport = async (request, response) => {
+    const {
+        reason,context,idAccount,accountReported
+    } = request.body;
+    const idAcountConverted  = mongoose.Types.ObjectId(idAccount);
+    const accountReportedConverted  = mongoose.Types.ObjectId(accountReported);
+    const dateNow = new Date();
+    const dateCreation = new Date(dateNow.getTime() - (dateNow.getTimezoneOffset() * 60000 )).toISOString().slice(0, 10);
+    const report = new Reports ({
+        reason: reason,
+        context:context,
+        dateCreation: dateCreation,
+        idAccount: idAcountConverted,
+        accountReported:accountReportedConverted
+    });
+    await report.save()
+    .then(function (report) {  
+        response.status(StatusCodes.CREATED).json(report);
+    })
+    .catch(function (error){
+        responseServer(response, error);
+    });
+}
+
+const getReportsFilters = async (request, response) => {
+    const filter = request.params.filter;
+    const criterion = request.params.criterion;
+    if(filter == "dateCreation"){
+        Reports.find({dateCreation: criterion}, {dateCreation:1, idAccount:1, accountReported:1})
+        .populate({path: 'idAccount', select: 'name lastname -_id'})
+        .populate({path: 'accountReported', select: 'name lastname -_id'})
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Martha-Miroslava/APIBiziidiDiidxaza/pullRequest?prid=8267260)
```suggestion
        .populate({path: "accountReported", select: 'name lastname -_id'})
```